Geometry/HcalTowerAlgo/test/HcalGeometryPlan1Tester.cc

```diff
@@ -0,0 +1,81 @@
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "Geometry/HcalCommonData/interface/HcalDDDRecConstants.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/CaloGeometry/interface/CaloCellGeometry.h"
+#include "Geometry/HcalTowerAlgo/interface/HcalGeometry.h"
+#include "Geometry/HcalTowerAlgo/interface/HcalFlexiHardcodeGeometryLoader.h"
+#include "DataFormats/HcalDetId/interface/HcalDetId.h"
+#include <iostream>
+#include <string>
+
+class HcalGeometryPlan1Tester : public edm::one::EDAnalyzer<> {
+
+public:
+  explicit HcalGeometryPlan1Tester( const edm::ParameterSet& );
+  ~HcalGeometryPlan1Tester( void ) {}
+    
+  void beginJob() override {}
+  void analyze(edm::Event const&, edm::EventSetup const&) override;
+  void endJob() override {}
+
+private:
+  edm::ParameterSet ps0_;
+  bool              geomDB_;
+};
+
+HcalGeometryPlan1Tester::HcalGeometryPlan1Tester( const edm::ParameterSet& iConfig ) :
+  ps0_(iConfig) {
+  geomDB_ = iConfig.getParameter<bool>("GeometryFromDB");
+}
+
+void HcalGeometryPlan1Tester::analyze(const edm::Event& /*iEvent*/, 
+				      const edm::EventSetup& iSetup) {
+
+  edm::ESHandle<HcalDDDRecConstants> hDRCons;
+  iSetup.get<HcalRecNumberingRecord>().get(hDRCons);
+  const HcalDDDRecConstants hcons = (*hDRCons);
+  edm::ESHandle<HcalTopology> topologyHandle;
+  iSetup.get<HcalRecNumberingRecord>().get(topologyHandle);
+  const HcalTopology topology = (*topologyHandle);
+  HcalGeometry* geom(0);
+  if (geomDB_) {
+    edm::ESHandle<CaloGeometry> pG;
+    iSetup.get<CaloGeometryRecord>().get(pG);
+    const CaloGeometry* geo = pG.product();
+    geom = (HcalGeometry*)(geo->getSubdetectorGeometry(DetId::Hcal,HcalBarrel));
+  } else {
+    HcalFlexiHardcodeGeometryLoader m_loader(ps0_);
```

Comment:
@bsunanda - a possibility of an HcalGeometry creation outside of an EventSetup is a bad practice. I hope, it is limited to the test only.